notifications_utils/countries/data.py

```diff
@@ -0,0 +1,63 @@
+import json
+import os
+
+
+def _load_data(filename):
+    with open(
+        os.path.join(os.path.dirname(__file__), '_data', filename)
+    ) as contents:
+        if filename.endswith('.json'):
+            return json.load(contents)
+        return [line.strip() for line in contents.readlines()]
+
+
+def find_canonical(item, graph, key):
+    if item['meta']['canonical']:
+        return key, item['names']['en-GB']
+    return find_canonical(
+        graph[item['edges']['from'][0]],
+        graph,
+        key,
+    )
+
+
+# Copied from
+# https://github.com/alphagov/govuk-country-and-territory-autocomplete
+# /blob/b61091a502983fd2a77b3cdb5f94a604412eb093
+# /dist/location-autocomplete-graph.json
+_graph = _load_data('location-autocomplete-graph.json')
+
+UK = 'United Kingdom'
+
+COUNTRIES_AND_TERRITORIES = [
+    find_canonical(item, _graph, item['names']['en-GB'])
+    for item in _graph.values()
+]
+
+ADDITIONAL_SYNONYMS = [
+    (key, value)
+    for key, value in _load_data('synonyms.json').items()
+]
+
+_UK_ISLANDS_LIST = _load_data('uk-islands.txt')
+_EUROPEAN_ISLANDS_LIST = _load_data('european-islands.txt')
+
+UK_ISLANDS = [
+    (synonym, synonym) for synonym in _UK_ISLANDS_LIST
```

Comment:
It may not be a bad thing, but it's interesting to see that you are treating 'Isle of Man' (or similar islands) as not a synonym of the UK. That was my initial mental model. Instead you are treating `Isle of man` as it's own country and then the `UK_POSTAGE_REGIONS` decides if it's in the UK postage region. This doesn't match the pattern that you are doing for places like `Wales` which are treated as a synonym of `UK`. Any particular thoughts behind this?

Reply:
For parcels (but not letters) Royal Mail treat Jersey etc differently from the rest of the UK. Therefore they recognise it as a valid destination. The worry is that because these places are quite small, letters might only be addressed to the island, and not a town, for example this is a very deliverable letter:

> House Name
> Castle Road
> Guernsey

However if we treat _Guernsey_ as a synonym for _United Kingdom_, and then go on to strip it we get an ambiguous address:

> House Name
> Castle Road

The difference with Wales is that we want to treat it as a synonym for _United Kingdom_ then remove it, to ensure the last line is a postcode, so this:

> 1234 Castle Road
> Cardiff
> CF10 XYZ
> Wales

Becomes:
> 1234 Castle Road
> Cardiff
> CF10 XYZ

…and the we can check that _CF10 XYZ_ is a valid postcode.

But now I’m wondering after writing that why we aren’t just doing the same for Jersey etc… 🤔

Reply:
Another issue here is that Sark, another one of the channel islands, costs the same as sending to Guernsey<sup>1</sup> but isn’t on the list of places you can send mail to. So if we got a letter addressed to Sark should we:
- let it through
- rewrite it to Guernsey, because that’s where the post office is
- remove it, and require a postcode

1. https://personal.help.royalmail.com/app/answers/detail/a_id/98/~/help-sending-mail-to-the-channel-islands-and-isle-of-man

Reply:
Consulted with @yahoopete and decided to treat the Channel Islands and Isle of Man the same as the rest of the UK, ie requiring a postcode. Done as part of df50dc1.